staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/aggregator.go

```diff
@@ -120,7 +120,7 @@ func BuildAndRegisterAggregator(downloader *Downloader, delegationTarget server.
 	}
 
 	// Install handler
-	s.openAPIVersionedService, err = handler.NewOpenAPIService(specToServe)
+	s.openAPIVersionedService, err = handler.NewOpenAPIService(specToServe, s.GetNewOpenAPISpec)
```

Comment:
The OpenAPIService is generic and is used by not just the aggregated api server, but also the one for CRDs and built-ins. (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go#L42)

I think we probably only need the lazy aggregation for the aggregator, but we'll need to make sure our code doesn't conflict with the other api servers.

Reply:
It sounds like the other use cases for `OpenAPIService` should be (in theory) okay with the lazy aggregation code even if they don't actively benefit, though this will need to be double-checked with testing later on, like you said.
This does get me wondering whether or not the aggregator should have its own separate codebase, though.